src/monattack.cpp

```diff
@@ -782,6 +812,9 @@ void mattack::spit_sap(monster *z, int index)
 
 void mattack::triffid_heartbeat(monster *z, int index)
 {
+    if( z->friendly ) {
+        return; // TODO: handle friendly monsters
```

Comment:
So far as I'd imagine, a "friendly" triffid heart would indicate that the PC has been subsumed into the local grove and is no longer a free actor (GAME OVER press spacebar).  If one somehow friendlies the thing whilst retaining independence, it makes sure you can leave (open walls between it and the stairway).